performance-tests/perf-test-commits.sh

```diff
@@ -0,0 +1,86 @@
+#! /usr/bin/env bash
+#
+# Copyright Kroxylicious Authors.
+#
+# Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+#
+
+PERF_TESTS_DIR=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+
+source "${PERF_TESTS_DIR}"/../scripts/common.sh
+
+GREEN='\033[0;32m'
+NOCOLOR='\033[0m'
+
+FIND_COMMAND="find"
+if [ "$OS" = 'Darwin' ] && resolveCommand gfind ; then
+  FIND_COMMAND=gfind
+  ENABLE_REGEX="-regextype posix-extended"
+elif [ "$OS" = 'Darwin' ] ; then
+  # for MacOS
+  ENABLE_REGEX="-E"
+else
+  # for Linux and Windows
+  ENABLE_REGEX="-regextype posix-extended"
+fi
+
+COMMITS=( "$@" )
+
+#Cross platform temp directory creation based on https://unix.stackexchange.com/a/84980
+RESULTS_DIR=${RESULTS_DIR:=$(mktemp -d -t kroxyliciousPerfTestResults.XXXXXX 2>/dev/null || mktemp -d -t 'kroxyliciousPerfTestResults')}
+JSON_TEMP_DIR=${JSON_TEMP_DIR:=$(mktemp -d -t kroxyliciousPerfTestJson.XXXXXX 2>/dev/null || mktemp -d -t 'kroxyliciousPerfTestJson')}
+echo -e "Writing results to: ${GREEN}${RESULTS_DIR}${NOCOLOR}"
+
+export PUSH_IMAGE=y #remove this once pulling is optional to save some time.
+export TEMP_BUILD=y
+
+checkoutCommit() {
+  local COMMIT_ID=$1
+  echo -e "Checkout ${GREEN}${COMMIT_ID}${NOCOLOR}"
+  git checkout --quiet "${COMMIT_ID}"
+}
+
+buildImage() {
+  local COMMIT_ID=$1
+  echo -e "Building image with tag ${GREEN}g_${COMMIT_ID}${NOCOLOR}"
+  "${PERF_TESTS_DIR}/../scripts/build-image.sh" -t "g_${COMMIT_ID}" -s > /dev/null
+}
+
+runPerfTest() {
+  local COMMIT_ID=$1
+  export KIBANA_OUTPUT_DIR=${RESULTS_DIR}/${COMMIT_ID}
+  mkdir -p "${KIBANA_OUTPUT_DIR}"
+  export KROXYLICIOUS_IMAGE="${REGISTRY_DESTINATION}:g_${COMMIT_ID}"
+  echo -e "Running tests using ${GREEN}${KROXYLICIOUS_IMAGE}${NOCOLOR}"
+  "${PERF_TESTS_DIR}/perf-tests.sh"
+}
+
+mergeResults() {
+ mapfile -t TEST_NAMES < <(${FIND_COMMAND} "${RESULTS_DIR}" -type d ${ENABLE_REGEX} -regex  ".*\/[0-9]{2}-.*" -exec basename {} \; | sort | uniq)
```

Comment:
`mapfile` is bash 4.4 thing, which isn't available by default in Mac OS so we may need to do something else here but this is `shellcheck`'s suggested fix...